connectivity/FEATURE_BLE/source/generic/SecurityManagerImpl.h

```diff
@@ -19,62 +19,74 @@
 
-#include <stdint.h>
-#include "CallChainOfFunctionPointersWithContext.h"
+#include <cstdint>
+
 #include "platform/Callback.h"
 
-#include "ble/common/ble/BLETypes.h"
-#include "ble/common/ble/blecommon.h"
 #include "ble/Gap.h"
-
-#include "ble/common/ble/GapTypes.h"
-#include "ble/common/ble/BLETypes.h"
-#include "ble/internal/SecurityDb.h"
-#include "ble/internal/PalConnectionMonitor.h"
-#include "ble/internal/PalSigningMonitor.h"
-#include "ble/internal/PalSecurityManager.h"
 #include "ble/SecurityManager.h"
 
+#include "ble/common/CallChainOfFunctionPointersWithContext.h"
+#include "ble/common/BLETypes.h"
+#include "ble/common/blecommon.h"
+#include "ble/common/BLETypes.h"
+
+#include "source/pal/GapTypes.h"
+#include "source/pal/PalConnectionMonitor.h"
+#include "source/pal/PalSigningMonitor.h"
+#include "source/pal/PalSecurityManager.h"
+
+#include "source/generic/SecurityDb.h"
+
+
 namespace ble {
+class PalGenericAccessService;
 
-class BLEInstanceBase;
+namespace impl {
 
 class SecurityManager :
-    public ble::interface::SecurityManager,
     public ble::PalSecurityManagerEventHandler,
     public ble::PalConnectionMonitorEventHandler,
-    public ble::PalSigningMonitorEventHandler
-{
+    public ble::PalSigningMonitorEventHandler {
     friend class ble::PalConnectionMonitorEventHandler;
-    friend BLEInstanceBase;
+
     friend PalGenericAccessService;
     friend PalSecurityManager;
 
+    using SecurityIOCapabilities_t = ble::SecurityManager::SecurityIOCapabilities_t;
+    using SecurityMode_t = ble::SecurityManager::SecurityMode_t;
+    using SecurityManagerShutdownCallback_t = ble::SecurityManager::SecurityManagerShutdownCallback_t;
+    using SecurityManagerShutdownCallbackChain_t = ble::SecurityManager::SecurityManagerShutdownCallbackChain_t;
+    using EventHandler = ble::SecurityManager::EventHandler;
+    using Passkey_t  = ble::SecurityManager::Passkey_t ;
+
+    static auto constexpr IO_CAPS_NONE = ble::SecurityManager::IO_CAPS_NONE;
+
 public:
     ////////////////////////////////////////////////////////////////////////////
     // SM lifecycle management
     //
 
     ble_error_t init(
-	    bool                     enableBonding = true,
-        bool                     requireMITM   = true,
-        SecurityIOCapabilities_t iocaps        = IO_CAPS_NONE,
-        const Passkey_t          passkey       = NULL,
-        bool                     signing       = true,
-        const char              *dbFilepath    = NULL
+        bool enableBonding = true,
+        bool requireMITM = true,
+        SecurityIOCapabilities_t iocaps = IO_CAPS_NONE,
+        const Passkey_t passkey = nullptr,
+        bool signing = true,
+        const char *dbFilepath = nullptr
     );
 
-    ble_error_t setDatabaseFilepath(const char *dbFilepath = NULL);
+    ble_error_t setDatabaseFilepath(const char *dbFilepath = nullptr);
 
-    ble_error_t reset(void);
+    ble_error_t reset(ble::SecurityManager* sm);
```

Comment:
why is secruity manager passed in to a non static function of security manger?

Reply:
The body invoke a callbacks that needs a pointer to the `SecurityManager` seen by the user.